code/drv_flow/src/wattrex_driver_flow/drv_flow.py

```diff
@@ -4,78 +4,207 @@
+
+                # failed = command_rec.status == SysShdNodeStatusE.COMM_ERROR
+                # if failed:
+                #     self.__tx_chan.send_data(msg)
+                #     break
+                msg = command_rec.payload[0]
+                if len(msg) > 0 and ('ERROR' not in msg) and ('IDN' in msg):
+                    msg = msg.split(':')
+                    self.__device_id = int(msg[msg.index('DEVice')+1])
+                    self.__firmware_version = int(msg[msg.index('VERsion')+1])
+                    exception = False
+                else:
+                    msg = DrvScpiCmdDataC(data_type = DrvScpiCmdTypeE.WRITE_READ,
+                            port = self.__port, payload = ScpiCmds.READ_INFO.value)
+                    self.__tx_chan.send_data(msg)
+
+        self.__rx_chan.delete_until_last()
+        if exception:
+            raise ConnectionError("Device not found")
+
+
+    @property
+    def device_id(self) -> int:
+        ''' Get the device id. '''
+        return self.__device_id
+
+
+    @property
+    def firmware_version(self) -> int:
+        ''' Get the firmware version. '''
+        return self.__firmware_version
+
 
     def get_meas(self) -> DrvFlowDataC:
-        """ Get the measurement of the flowmeter"""
-        req_msg = (ScpiCmds.REQ_MEAS.value + '\n').encode()
-        self.serial.write(req_msg)
-
-        n_tries = 0
-        datos = None
-        while n_tries < _CONSTANTS.MAX_ATTEMPTS:
-            res = self.serial.readline()
-            print(res)
-            if res != b'':
-                datos = res.decode()
-                break
-            n_tries += 1
-
-        print(f"----------\n{datos}----------")
-        res = None
-        if datos is not None and datos.startswith(ScpiCmds.MEAS_DATA.value):
-            flows = datos.split(' ')
-            print(f"Data received: {flows}")
-            res = DrvFlowDataC(int(flows[1]), int(flows[2]))
-        return res
-
-    def close(self):
-        """ Close the serial port"""
-        self.serial.close()
+        ''' Get the measurement of the flowmeter.
+        Args:
+            - None.
+        Returns:
+            - res (DrvFlowDataC): Get the measurement of the flowmeter.
+        Raises:
+            - None.
+        '''
+        if not self.__wait_4_response:
+            msg = DrvScpiCmdDataC(data_type = DrvScpiCmdTypeE.WRITE_READ,
+                                port = self.__port,
+                                payload = ScpiCmds.GET_MEAS.value + '\n')
+            log.info(f"Message to send: {msg.payload}")
+            self.__tx_chan.send_data(msg)
+            self.__wait_4_response = True
+        else:
+            if not self.__rx_chan.is_empty():
+                data : DrvScpiCmdDataC = self.__rx_chan.receive_data()
+                msg_received = data.payload[0]
+                log.critical(f"Message received: {msg_received}")
+                msg_received = msg_received.split(':')

```

Comment:
Split message after checking length of msg_received

Reply:
Done.